canlink/canclient.go

```diff
@@ -186,3 +188,7 @@ func (c *CanClient) StopTracking() (map[uint32]int, error) {
 func (c *CanClient) IsTracking() bool {
 	return c.tracking
 }
+
+func isIdNotInDatabaseError(err error) bool {
```

Comment:
This function is unnecessary. The code where it is used should be reorganized because the boolean logic is pretty messy

```go
if err != nil && !isIdNotInDatabaseError(err)
// expands to
if err != nil && !(err != nil AND String.contains(...))
// equiv to
if err != nil && (err == nil OR !String.contains(...))
// equiv to
if err != nil && !String.contains(...)
```
So I think the error checking statements should be restructured to
```go
if err != nil { // this follows regular go style
    if strings.Contains(err.Error(), _idNotInDatabaseErrorIndicator) {
		c.l.Debug("found a message we do not recognize")
		// Here we have simply read a can frame that we do not know how to unmarshal, continue to next frame.
		continue
    } else {
        return nil, errors.Wrap(err, "unmarshal frame")
    }
}
```

This avoids an extra function definition, making it easier for the reader to understand and makes the error check follow the Go Style of `if err != nil { do something }`.

I also argue against the function name helping understand because the `_idNotInDatabaseErrorIndicator` constant is very well named so the if statement is very clear.